README.md

```diff
@@ -0,0 +1,722 @@
+2. Run `gcloud config set project PROJECT_ID`. Where PROJECT_ID is the project
+   you want to use. The Snapshot Debugger CLI always acts on the current
+   `gcloud` configured project.
+
+## Enable Firebase for your Google Cloud Project
+
+The Snapshot Debugger CLI and agents use a Firebase Realtime Database (RTDB) to
+communicate.
+
+If you already use Firebase in your project, skip to the [Set up the Firebase
+RTDB](#set-up-the-firebase-rtdb) section.
+
+1. Add Firebase to your project:
+
+   https://console.firebase.google.com/?dlAction=MigrateCloudProject&cloudProjectNumber=PROJECT_ID
+
+   Where PROJECT_ID is your project ID
+
+2. Select your
+   [Project ID](https://cloud.google.com/resource-manager/docs/creating-managing-projects#identifying_projects)
+    and click **Continue**. If you have billing enabled in your project, the
+    pay-as-you-go Blaze plan is selected, otherwise the free Spark plan is
+    selected. If you have billing enabled and want to use the free Spark plan,
+    set up a new project without billing enabled.
+
+   Note: The Snapshot Debugger uses the Firebase RTDB service. Most users' usage
+   will be low enough to remain under the [free usage
+   limits](https://firebase.google.com/pricing?).
+
+3. If you are using the pay-as-you-go Blaze plan, click **Confirm plan**. You
+   are not prompted to confirm if you are on the free Spark plan.
+
+4. Read the information under **A Few things to remember when adding Firebase to
+   a Google Cloud project** then click **Continue**.
+
+5. Toggle the **enable analytics** option to enable or disable Google Analytics
+   for Firebase. Google Analytics isn't required for Debugger use.
+
+6. Click **Continue**.
+
+7. Click **Get Started**.
+
+8. Click on your project.
+
+### Set up the Firebase RTDB
+
+The instructions are slightly different depending on whether you are on the
+Spark or Blaze billing plan. Follow the steps in the following section for the
+plan you have.
+
+You can check what billing plan is in effect for your project on the Firebase Usage & Billing page:
+
+https://console.firebase.google.com/project/PROJECT_ID/usage/details
+
+Where PROJECT_ID is your project ID
+
+#### Blaze plan RTDB setup
+
+This will instruct the debugger CLI to create and use a database with the name
+`PROJECT_ID-cdbg`
+
+1. Run `python3 cli/src/cli.py init` in the cloned `snapshot-debugger`
+   directory.
+2. The output resembles the following:
+
+```
+Project 'test-proj' is successfully configured with the Firebase Realtime
+Database for use by Snapshot Debugger.
+
+The full database information is below. If you have specified a custom database
+ID the url below is the one you'll need to specify when using the other cli
+commands.
+
+  name:         projects/23498723497/locations/us-central1/instances/test-proj-cdbg
+  project:      projects/23498723497
+  database url: https://test-proj-cdbg.firebaseio.com
+  type:         USER_DATABASE
+  state:        ACTIVE
```

Comment:
It would be so nice if there were a way to tie together the README content and the string constants in the scripts...

Reply:
agreed.